lib/src/components/NxList/__tests__/NxList.test.tsx

```diff
@@ -4,192 +4,216 @@
-  async function mountAttached(jsx: ReactElement) {
-    let retval: ReactWrapper;
-    await act(async () => {
-      retval = mount(jsx, { attachTo: getMountContainer() });
-    });
-    retval!.update();
+    expect(view.getByRole('list')).toBeInTheDocument();
+  });
 
-    return retval!;
-  }
+  it('sets the specified classnames in addition to the defaults', async function() {
+    const el = renderEl({ className: 'foo' }),
+        defaultEl = renderEl()!;
 
-  afterEach(function() {
-    mountContainers.forEach(container => document.body.removeChild(container));
-    mountContainers = [];
-  });
+    await runTimers();
 
-  const minimalProps: NxListProps = {};
-  const getShallow = getShallowComponent(NxList, minimalProps);
+    expect(el).toHaveClass('foo');
 
-  it('renders a list', function() {
-    const nxList = getShallow();
-    expect(nxList).toMatchSelector('.nx-list');
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
-  it('renders the classNames given to it', function() {
-    const extendedProps: Partial<NxListProps> = {
-      className: 'test-classname ufo'
-    };
-    const nxList = getShallow(extendedProps);
-    expect(nxList).toMatchSelector('.nx-list.test-classname.ufo');
-  });
+  it('sets the specified attrs', async function() {
+    const el = renderEl({ id: 'foo', lang: 'en' });
 
-  it('renders a bulleted list correctly', function() {
-    const contentEl = getShallow({bulleted: true});
-    expect(contentEl).toHaveClassName('nx-list--bulleted');
-  });
-  it('shows the emptyMessage when there are no children', async function() {
-    const component = await mountAttached(
-      <NxList emptyMessage="Empty message">
-      </NxList>
-    );
+    await runTimers();
 
-    expect(component.find('span')).toHaveText('Empty message');
+    expect(el).toHaveAttribute('id', 'foo');
+    expect(el).toHaveAttribute('lang', 'en');
   });
 
-  it('shows the emptyMessage when the children are removed after the existing', async function() {
-    const component = await mountAttached(
-      <NxList emptyMessage="Empty message">
-        <NxList.Item>
-          <NxList.Text>Test 1</NxList.Text>
-        </NxList.Item>
-        <NxList.Item>
-          <NxList.Text>Test 2</NxList.Text>
-        </NxList.Item>
-      </NxList>
-    );
-
-    await act(async () => {
-      component.setProps({ children: [] });
-    });
-    component.update();
+  it('sets a ref to the element', async function() {
+    const ref = React.createRef<HTMLUListElement>(),
+        el = renderEl({ ref });
 
-    expect(component.find('span')).toHaveText('Empty message');
+    await runTimers();
+
+    expect(ref.current).toBe(el);
   });
 
-  it('shows the emptyMessage when there are no children, no error, and not loading', async function() {
+  it('renders children correctly', function() {
+    const children = <NxList.Item>Cat</NxList.Item>,
```

Comment:
Render more than one child, just to be safe

Reply:
Updated here d48f9f648f10da27f8a30b98d9ab933ec292b005